app/models/manageiq/providers/openstack/cloud_manager/vm/associate_ip.rb

```diff
@@ -0,0 +1,47 @@
+module ManageIQ::Providers::Openstack::CloudManager::Vm::AssociateIp
+  extend ActiveSupport::Concern
+
+  included do
+    supports :associate_floating_ip do
+      if cloud_tenant.nil? || Rbac.filtered(cloud_tenant.floating_ips).empty?
```

Comment:
Rbac itself handles Vm's rbac.
If you want ip addresses then add that as a class in rbac.

This class should not need to handle this.
A bit with tenancy is in the model itself.

Do you have rbac on ip addresses? or on the VM's itself?